semgrep-core/src/matching/SubAST_generic.ml

```diff
@@ -32,6 +32,11 @@ let go_really_deeper_stmt = ref true
 (* Sub-expressions and sub-statements *)
 (*****************************************************************************)
 
+let subexprs_of_any_list xs = xs |> List.fold_left (fun x -> (function
+  | E e -> e :: x
+  | _ -> x
+)) []
```

Comment:
Your use of fold_left is weird to me. I would write it as:
```ocaml
let subexprs_of_any_list anys =
  List.fold_left (fun acc any ->
    match any with
      | E e -> e :: acc
      | _ -> acc
  ) [] anys
```
The part that's weird to me in your version is to not have the two arguments side by side.

Reply:
^^ yeah looks better